File: src/main/java/de/esymetric/jerusalem/ownDataRepresentation/geoData/Coord2.java
package de.esymetric.jerusalem.ownDataRepresentation.geoData;

public class Coord2
{
    public double x;
    public double y;

    public Coord2() { }

    public Coord2(double x, double y)
    {
        this.x = x;
        this.y = y;
    }

    public double distance(Coord2 c)
    {
        double dx = c.x - x;
        double dy = c.y - y;
        return Math.sqrt(dx * dx + dy * dy);
    }
}

File: src/main/java/de/esymetric/jerusalem/ownDataRepresentation/geoData/Coord2.kt
package de.esymetric.jerusalem.ownDataRepresentation.geoData

class Coord2 {
    var x = 0.0
    var y = 0.0

    constructor() {}
    constructor(x: Double, y: Double) {
        this.x = x
        this.y = y
    }

    fun distance(c: Coord2): Double {
        val dx = c.x - x
        val dy = c.y - y
        return Math.sqrt(dx * dx + dy * dy)
    }
}